src/lavinmq/http/http_server.cr

```diff
@@ -80,6 +82,20 @@ module LavinMQ
         @http.try &.close
         File.delete?(INTERNAL_UNIX_SOCKET)
       end
+
+      # Starts a HTTP server that binds to the internal UNIX socket used by lavinmqctl.
+      # The server returns 503 to signal that the node is a follower and can not handle the request.
+      def self.follower_internal_socket_http_server
+        http_server = ::HTTP::Server.new do |context|
+          context.response.status_code = 503
+          context.response.status_message = "This node is a follower and does not handle HTTP requests."
+        end
+        self.bind_internal_unix(http_server)
```

Comment:
it makes sense to reuse `bind_internal_unix` but it also leaves the other method somewhat strange, for example `listen`  uses `@http` but that isn't the "correct" server when we are in follower mode. 

I think this works but makes it harder to track which "http server" we are talking about. So I would suggest copy and inline the lines from `bind_internal_unix` into the follower method.  That way we dont need to change the signature of the `bind_internal_unix` either

Reply:
Yeah, I was trying not to duplicate code, but maybe you're right. It feels a little messy mixing as it is now.